scripts/visualize/visualize.R

```diff
@@ -33,3 +33,31 @@ visualize.qTDS <- function(viz){
   plot(TDS ~ Q, CuyahogaShort, log = "xy", col=colors$points)
   dev.off()
 }
+
+visualize.dots <- function(viz){
+  svg_text <- '<?xml version="1.0" encoding="UTF-8"?>
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 600 450" preserveAspectRatio="xMidYMid meet" version="1.1">
+  <g>
+  <circle cx = "400" cy = "100" r = "10" onmousemove = "hovertext(\'this circle is green\',evt)" onmouseout = "hovertext(\' \')" style="fill:green"/>
+  <rect x = "550" y = "20" height="410" width="40" style="fill:green"/>
+  <rect x = "505" y = "20" height="410" width="40" onmousemove = "hovertext(\'this mouser is below the tooltip\',evt)" onmouseout = "hovertext(\' \')" style="fill:grey"/>
+  <rect x = "40" y = "320" height="14" width="400" onmousemove = "hovertext(\'this shape happens to be blue\',evt)" onmouseout = "hovertext(\' \')" style="fill:blue"/>
+  <circle cx = "10" cy = "120" r = "8" onmousemove = "hovertext(\'this circle is red\',evt)" onmouseout = "hovertext(\' \')" style="fill:red"/>
+  </g>
+  <g id = "tooltip-group"/>
+  <rect x = "550" y = "20" height="410" width="40" onmousemove = "hovertext(\'this mouser is above the tooltip\',evt)" onmouseout = "hovertext(\' \')" style="fill-opacity:0"/>
```

Comment:
what does "mouser" mean? googling it brings up an electronics company and the cat at 10 Downing Street (and [this](http://www.urbandictionary.com/define.php?term=mouser)), and googling "mouser javascript" only brings up mouse function pages that never say "mouser".

i'm also confused by the text you chose to put in the hovertext. i thought maybe these bars would illustrate that the tooltip can go above or below the mouse pointer...but they don't...

Reply:
mouser is what you call a cat that is good at catching mice, or an svg element that contains the mouseover functions. Do we have a better name? that's just what I call them. Open to changing

This text means the element triggering the hovertext call is either above or below the tooltip group in the z-index of the page, not the y coordinates. If you play with this, you'll notice that moving the mouse quickly up the right rectangle is fluid, but doing the same on the grey rectangle causes the tooltip to flicker. This is because the mouse location "chased" the tooltip fast enough to be on top of it instead of the rectangle, so the tip briefly shuts off. Using a transparent element that just deals with the mouse events (not style) allows us to avoid that flickering, so I wanted to show this.

Reply:
with the definition, "mouser" is sensible and humorous - the svg element that catches the mouse.  maybe we just need to keep explaining it to people who ask (like me here) and make it common team knowledge. here's a not-too-different but not-exactly-the-same definition for a javascript class someone wrote: http://www.kevlindev.com/gui/utilities/mouser/index.htm

Re: over and under: ooooohhhhhh! definitely hadn't spotted that. thanks for the explanation.